### PR TITLE
Fix undefined properties on back-office pledge form

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix (some) undefined properties on back-office pledge form

Before
----------------------------------------
Several undefined properties being set / accessed

After
----------------------------------------
All undefined variables in the form removed (mostly using local variables instead)

Technical Details
----------------------------------------
The `_context` property is declared as protected in some places & public in others but the `context` property is on the main `CRM_Core_Form` class & has public getter / setters. I't so ubiquittous I'm temped to set it in `CRM_Core_Form::preProcess` - but this doesn't do that

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contact/Form/ContactFormTrait.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  public function getContactValue($fieldName) {
+    if ($this->isDefined('Contact')) {
+      return $this->lookup('Contact', $fieldName);
+    }
+    $id = $this->getContactID();
+    if ($id) {
+      // GetContactID may or may not have defined this.
+      if (!$this->isDefined('Contact')) {
+        $this->define('Contact', 'Contact', ['id' => $id]);
+      }
```
Comment (colemanw) [inline CRM/Contact/Form/ContactFormTrait.php:34]: This looks wrong @eileenmcnaughton - GetContactID should definitely have defined it.

Comment (colemanw) [reply CRM/Contact/Form/ContactFormTrait.php:34]: Although @eileenmcnaughton I feel like GetContactID probably shouldn't. Feels like overloading a getter function to also be a setter isn't right. I think this function should define the contact if not defined, and GetContactID should only do what it says & get the id.

Comment (eileenmcnaughton) [reply CRM/Contact/Form/ContactFormTrait.php:34]: @colemanw yeah - but that is what the function I moved is doing - I will revist once this is merged

Comment (eileenmcnaughton) [reply CRM/Contact/Form/ContactFormTrait.php:34]: actually maybe not - hmm - just pondering. Yeah I will remove

File: CRM/Contact/Form/ContactFormTrait.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+   *
+   * Override this for more complex retrieval as required by the form.
+   *
+   * @return int|null
+   *
+   * @noinspection PhpUnhandledExceptionInspection
+   * @noinspection PhpDocMissingThrowsInspection
+   */
+  public function getContactID(): ?int {
+    $id = (int) CRM_Utils_Request::retrieve('cid', 'Positive', $this);
```
[the remark this answers is not in the document]
Comment (eileenmcnaughton) [reply CRM/Contact/Form/ContactFormTrait.php:48]: I guess it doesn't need to be transparent - the point is `getContactID` should work reliably at all stages of submission

**Comment by colemanw** [comment]
Ok cool this looks good.

**Pull request merged**